### PR TITLE
Monica/searchroutes

### PR DESCRIPTION
**Pull request by myo3**

### CODE REVIEW

File: TCAT/Cells/RouteTableViewCell.swift [hunk 1]
```diff
@@ -173,10 +187,22 @@ class RouteTableViewCell: UITableViewCell {
...
             busIcons[i].center.y = (stopDots[i].center.y + stopDots[i+1].center.y)/2
-            busIcons[i].center.x = spaceXFromSuperviewLeft + (busIcons[i].frame.width/2)
+            if let walkIcon = busIcons[i] as? UIImageView{
+                walkIcon.center.x = spaceXFromSuperviewLeft + (busIcons[i-1].frame.width/2)
+                //create walk lines
+                let topDot = stopDots[i]
+                let bottomDot = stopDots[i+1]
+                let line = DottedLine(frame: CGRect(x: 0, y: topDot.frame.maxY, width: 1.0, height: bottomDot.frame.minY - topDot.frame.maxY))
+                line.backgroundColor = .white
+                print(line.frame.height)
```
Comment (mattbarker016) [inline TCAT/Cells/RouteTableViewCell.swift]: can remove or comment out

File: TCAT/Controllers/OptionsViewController.swift [hunk 2]
```diff
@@ -242,35 +294,74 @@ class OptionsViewController: UIViewController, UITableViewDelegate, UITableViewD
...
+        for route in routes{
+            var validRoute = true
+            let directions = route.directions
+            //Check directions to invalidate route
+            for i in 0..<directions.count{
+                if let walkDir = directions[i] as? WalkDirection{
+                    if i == 0{
+                        walkDir.calculateWalkingDirections({ (distance, walkTimeInterval) in
+                            //this might be sketch for leave now, check logic
+                            if self.searchTimeType == .leaveat{ //make sure if walk now to stop, get there before leaveat time
```
Comment (mattbarker016) [inline TCAT/Controllers/OptionsViewController.swift]: I mentioned in Austin's PR, but ideally these enums could be camel-cased (.leaveAt)

File: TCAT/Controllers/OptionsViewController.swift [hunk 3]
```diff
@@ -242,35 +294,74 @@ class OptionsViewController: UIViewController, UITableViewDelegate, UITableViewD
...
+                                let walkToStopDate = Date().addingTimeInterval(walkTimeInterval)
+                                if(walkToStopDate > self.searchTime!){
+                                    validRoute = false
+                                }else{
+                                    route.departureTime.addTimeInterval(-walkTimeInterval)
```
Comment (mattbarker016) [inline TCAT/Controllers/OptionsViewController.swift:314]: why is this negative?

**Review by mattbarker016 — COMMENTED** [review]
One errant negative sign, some tweaks

File: TCAT/Controllers/OptionsViewController.swift (hunk 3 above)
Comment (myo3) [reply TCAT/Controllers/OptionsViewController.swift:314]: b/c Shiv said he'll put the departureTime as the time the bus leaves. So to get the actual departure time I do: time bus leaves -  the time it takes to walk to the bus stop

**Pull request merged**